### PR TITLE
Create new chunker for composite key

### PR DESCRIPTION
**Pull request by YinanLiang**
Part 1 of fix https://github.com/squareup/spirit/issues/48

Use a different chunker for composite key. The new chunker currently does exactly the same thing as the universal chunker.

### CODE REVIEW

File: pkg/table/chunker_composite.go [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+	// particularly for the test suite.
+	DisableDynamicChunker bool // optimization to adjust chunk size.
+
+	// This is used for restore.
+	watermark             *Chunk
+	watermarkQueuedChunks []*Chunk
+
+	// The chunk prefetching algorithm is used when the chunker detects
+	// that there are very large gaps in the sequence.
+	chunkPrefetchingEnabled bool
```
Comment (morgo) [inline pkg/table/chunker_composite.go]:
```suggestion
```

File: pkg/table/chunker_composite.go [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+
+	minVal := t.chunkPtr
+	maxVal := t.chunkPtr.Add(incrementSize)
+	t.chunkPtr = maxVal
+	return &Chunk{
+		ChunkSize:  t.chunkSize,
+		Key:        key,
+		LowerBound: &Boundary{minVal, true},
+		UpperBound: &Boundary{maxVal, false},
+	}, nil
```
Comment (morgo) [inline pkg/table/chunker_composite.go]:
```suggestion
	return t.nextChunk()
```

File: pkg/table/chunker_composite.go [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+	return uint64(math.Ceil(float64(increment) / divideBy)) // ceil to guarantee at least 1 for progress.
+}
+
+func (t *chunkerComposite) logicalRange() uint64 {
+	if !t.Ti.maxValue.IsNil() && !t.Ti.minValue.IsNil() {
+		return t.Ti.maxValue.Range(t.Ti.minValue)
+	}
+	// For binary types, we can't really do anything useful yet:
+	return math.MaxUint64
+}
```
Comment (morgo) [inline pkg/table/chunker_composite.go]:
```suggestion
```

File: pkg/table/chunker_composite.go [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+	// - This new target wants to go higher
+	// - our current p90 is only a fraction of our target time
+	if t.chunkSize == MaxDynamicRowSize && newTargetRows > MaxDynamicRowSize && (p90 < targetTime*5) {
+		t.logger.Warnf("dynamic chunking is not working as expected: target-time=%s p90-time=%s new-target-rows=%d max-dynamic-row-size=%d",
+			time.Duration(targetTime), time.Duration(p90), uint64(newTargetRows), MaxDynamicRowSize,
+		)
+		t.logger.Warn("switching to prefetch algorithm")
+		t.chunkSize = StartingChunkSize // reset
+		t.chunkPrefetchingEnabled = true
+	}
```
Comment (morgo) [inline pkg/table/chunker_composite.go]:
```suggestion
```

Comment (YinanLiang) [reply pkg/table/chunker_composite.go]: That makes sense. Thanks Morgan. I was planning to remove the dynamic chunking logics in the next PR to make sure this PR doesn't change the current behaviour before we introduce more tests. But thinking about it again it makes sense to start with a minimum chunker. Removed those logic as suggested :D

File: pkg/migration/runner_test.go [hunk 5]
```diff
@@ -263,7 +262,7 @@ func TestTableLength(t *testing.T) {
 		Password: cfg.Passwd,
 		Database: cfg.DBName,
 		Threads:  16,
-		Table:    "thisisareallylongtablenamethisisareallylongtablename64characters",
+		Table:    "thisisareallylongtablenamethisisareallylongtablename60charac",
```
Comment (YinanLiang) [inline pkg/migration/runner_test.go:265]: I think this is the fix yeah?

Comment (morgo) [reply pkg/migration/runner_test.go:265]: In MySQL it's fixed at 4 chars, but since we need a _new table it requires an additional 5 chars.

**Pull request merged**